### PR TITLE
Collapse @ chains in SMT2 printer

### PR DESCRIPTION
**Pull request by HanielB**
This is useful for generating, from THF benchmarks, benchmarks that can be parsed by CVC4 in the HOL extension of SMT2

### CODE REVIEW

**Comment by ajreynol** [comment]
I am wondering if this should be a feature of the printer.  It seems that this should be somewhere else (a preprocessing pass / a script?)  I am a bit wary when the printer is deceiving the user what the node itself is.

**Comment by HanielB** [comment]
Maybe it'd be better if it were an option for the printer? The structure of the term does not affect the solver, only the parser. The use case of this for now is just to generate problems that CVC4 can parser.

**Comment by ajreynol** [comment]
If CVC4 can't parse the output, shouldn't we extend the parser instead of the printer?

**Comment by HanielB** [comment]
I think we should do both. :) It's a useful feature to normalize the output, but indeed it should not be the default case. So should we settle on opening an issue to extend the hosmt parser and to add an option to print normalized chains?

File: src/printer/smt2/smt2_printer.cpp [hunk 1]
```diff
@@ -509,7 +509,30 @@ void Smt2Printer::toStream(std::ostream& out,
   // uf theory
   case kind::APPLY_UF: typeChildren = true; break;
   // higher-order
-  case kind::HO_APPLY: break;
+  case kind::HO_APPLY:
+    // collapse "@" chains, i.e.
+    //
+    // @(@(a, b), c) --> (@ a b c)
```
Comment (ajreynol) [inline src/printer/smt2/smt2_printer.cpp]: This isn't the actual smt2 syntax right? Can you comment on what is actually being printed?

Comment (HanielB) [reply src/printer/smt2/smt2_printer.cpp]: the syntax for the above example would be 

`((a b) c)` becomes `(a b c)`

Comment (ajreynol) [reply src/printer/smt2/smt2_printer.cpp]: I mean, comment in the code :)

Comment (HanielB) [reply src/printer/smt2/smt2_printer.cpp]: ah, right hahaha will do

**Pull request merged**